### PR TITLE
Update HAL STM32

### PR DESCRIPTION
**Pull request by gautierg-st**
Update HAL STM32:
- C0: 1.1.0 -> 1.2.0
- F1: 1.8.5 -> 1.8.6
- F2: 1.9.4 -> 1.9.5
- F4: 1.28.0 -> 1.28.1
- G4: 1.5.2 -> 1.6.0
- H5: 1.2.0 -> 1.3.0
- H7RS: 1.0.0 -> 1.1.0
- U5: 1.5.0 -> 1.6.0
- WB: 1.19.1 -> 1.20.0

### CODE REVIEW

File: stm32cube/stm32f2xx/README [hunk 1]
```diff
@@ -35,6 +35,7 @@ License Link:
    https://opensource.org/licenses/BSD-3-Clause
 
 Patch List:
+--> please check that the following list is still valid:
```
Comment (erwango) [inline stm32cube/stm32f2xx/README:38]: Remove this line

File: stm32cube/stm32h7rsxx/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,5 @@
 # Copyright (c) 2019  Linaro Limited
 # Copyright (c) 2020 STMicroelectronics
-# Copyright (c) 2024 STMicroelectronics
```
Comment (erwango) [inline stm32cube/stm32h7rsxx/CMakeLists.txt:3]: Don't remove

File: stm32cube/stm32u5xx/README [hunk 3]
```diff
@@ -44,11 +44,6 @@ Patch List:
     -Added stm32cube/stm32u5xx/drivers/include/stm32_assert.h
     -Removed unused stm32cube/stm32u5xx/drivers/include/stm32_assert_template.h
 
-   *Fix GFXTIM HAL: Compilation issue
-    -"#if defined (GFXTIM)" is missing from drivers/include/stm32u5xx_hal_gfxtim.h
-   Impacted files: drivers/include/stm32u5xx_hal_gfxtim.h
-   Internal reference: 157095
```
Comment (erwango) [inline stm32cube/stm32u5xx/README:50]: I don't see this in the changes, was it delivered earlier ?

File: lib/stm32wb/hci/README [hunk 4]
```diff
@@ -60,6 +60,7 @@ License Link:
    opensource.org/licenses/BSD-3-Clause
 
 Patch List:
+--> please check that the following list is still valid:
```
Comment (erwango) [inline lib/stm32wb/hci/README:63]: Remove this line

File: stm32cube/common_ll/README.rst [hunk 5]
```diff
@@ -10,27 +10,27 @@ below.
...
 stm32l4xx       1.18.1
 stm32l5xx       1.5.1
 stm32mp1xx      1.6.0
 stm32u0xx       1.1.0
-stm32u5xx       1.5.0
+stm32u5xx       1.6.0
 stm32wb0x       1.0.0
-stm32wbaxx      1.3.1
-stm32wbxx       1.19.1
+stm32wbaxx      1.4.1
```
Comment (erwango) [inline stm32cube/common_ll/README.rst:33]: Don't update wba

**Review by erwango — CHANGES_REQUESTED** [review]

File: stm32cube/stm32f2xx/README (hunk 1 above)
Comment (gautierg-st) [reply stm32cube/stm32f2xx/README]: Done

File: stm32cube/stm32h7rsxx/CMakeLists.txt (hunk 2 above)
Comment (gautierg-st) [reply stm32cube/stm32h7rsxx/CMakeLists.txt:3]: Done

File: lib/stm32wb/hci/README (hunk 4 above)
Comment (gautierg-st) [reply lib/stm32wb/hci/README]: Done

File: stm32cube/common_ll/README.rst (hunk 5 above)
Comment (gautierg-st) [reply stm32cube/common_ll/README.rst]: Done

File: stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h [hunk 6]
```diff
@@ -333,8 +334,8 @@
 #endif /* HAL_ETH_MODULE_ENABLED */
 
 #ifdef HAL_ETH_LEGACY_MODULE_ENABLED
-  #include "Legacy/stm32f4xx_hal_eth_legacy.h"
-#endif /* HAL_ETH_MODULE_ENABLED */
+  #include "stm32f4xx_hal_eth_legacy.h"
+#endif /* HAL_ETH_LEGACY_MODULE_ENABLED */
```
Comment (erwango) [inline stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:338]: Although we don't use this file we should probably follow this change for consistency.

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**